pygeos/test/test_strtree.py

```diff
@@ -0,0 +1,75 @@
+import pygeos
+from pygeos import box
+import pytest
+import numpy as np
+import sys
+from contextlib import contextmanager
+from .common import point, empty
+
+
+@contextmanager
+def assert_increases_refcount(obj):
+    before = sys.getrefcount(obj)
+    yield
+    assert sys.getrefcount(obj) == before + 1
+
+
+@contextmanager
+def assert_decreases_refcount(obj):
+    before = sys.getrefcount(obj)
+    yield
+    assert sys.getrefcount(obj) == before - 1
+
+
+@pytest.fixture
+def tree():
+    geoms = pygeos.points(np.arange(10), np.arange(10))
+    yield pygeos.STRtree(geoms)
+
+
+def test_init_with_none():
+    pygeos.STRtree(np.array([None]))
+
+
+def test_init_increases_refcount():
+    arr = np.array([point])
+    with assert_increases_refcount(point):
+        _ = pygeos.STRtree(arr)
+
+
+def test_del_decreases_refcount():
+    arr = np.array([point])
+    tree = pygeos.STRtree(arr)
+    with assert_decreases_refcount(point):
+        del tree
+
+
+def test_query_no_geom(tree):
+    with pytest.raises(TypeError):
+        tree.query("I am not a geometry")
+
+
+def test_query_none(tree):
+    with pytest.raises(TypeError):
+        tree.query(None)
```

Comment:
Should we return an empty array instead? Or have an option for it? 
Just thinking aloud (it might be useful, as you typically need to handle the empty case anyway, as valid geometries can also return that)

Reply:
Agreed, see above. As `pygeos.intersects(geometry, None) == False`, I think returning an empty list here would indeed make more sense.